### PR TITLE
Update header component

### PR DESCRIPTION
**Pull request by nonword**
Principally, this updates header component to latest. Per repo stated merge strategy, this feature has been merged and deployed to development:

http://nypl-global-search-development.us-east-1.elasticbeanstalk.com/search/toast/

This PR also improves a few non-code things:
 - Adds `.env-sample` as a template for `.env` and with added detail about variable purpose.
 - Improves/fixes README coverage of env vars - especially with respect to what's required and what the effective defaults are
 - Improved README with respect to merge strategy
 - Adds `.nvmrc` to match production Node version

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,38 +18,50 @@ $ npm install
 From the CLI, run `npm run lint` to run the ESLint tool for checking JavaScript syntax.  Here is an example of how to lint only one file: `npm run lint-single-file src/app/components/TabItem/TabItem.jsx`
 
 ### Environment Variables
-We use four environment variables so far for this application.
+
+The following environment variables are used by this application:
 
   - `API_ROOT` indicates the root of the endpoint this app calls for search requests.
-  - `APP_ENV` indicates the environment the app is running in. It could be `development`, `qa`, or `production`. The default value is `production`.
+  - `APP_ENV` indicates the environment the app is running in. It could be `development`, `qa`, or `production`. If left unset, the app mostly behaves the same as "production".
```
Comment (danamansana) [inline README.md]: yikes. Do you know any specific instances where it behaves differently from "production"?

File: .env-sample [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export NODE_ENV=development # This is effective default. May also be production
+export REGION_ENV=us-east-1 # This is default
+
+# Optionally override Google Search API endpoint (normally determined by APP_ENV):
+# export API_ROOT=https://www.googleapis.com/customsearch/v1?key=SUPERSECRETGOOGLEAPIKEYS
+
+# The following effectively defaults to nypl-sandbox if APP_ENV is
+# 'development'. Defaults to nypl-digital-dev otherwise.
+# export AWS_PROFILE=
```
Comment (danamansana) [inline .env-sample]: 👍 for comments in sample

File: README.md (hunk 1 above)
Comment (nonword) [reply README.md]: `APP_ENV` is there to establish defaults for Google Search API and Redis (for caching). If `APP_ENV` is not set, the Google Search API effectively defaults to the `production`. However, the Redis client doesn't default to the production redis (which is, I think, *good*). It defaults to `localhost` (i.e. you have to have Redis running locally in another process). If you disable caching via `SKIP_CACHING=true`, an unset `APP_ENV` behaves identically to one set to `production`. This kind of confusion may be why 12-factor app [recommends](https://12factor.net/config) against bundling config by deployment; It immediately gets complicated/restrictive.

Comment (danamansana) [reply README.md]: Ok well it might be nice to give an example case where this makes a difference in the README if you have time. Approving.

**Pull request merged**